pkg/agent/stream/watchtopic.go

```diff
@@ -92,248 +132,72 @@ func getCachedWatchTopic(c cache.GetItem, key string) (*mv1.WatchTopic, error) {
 	return v, nil
 }
 
+type kindValues struct {
+	KindName   string
+	EventTypes []string
+	Group      string // blank defaults to management in template
+	ScopeKind  string // blank defaults to Environment in template
+	ScopeName  string // blank generates no scope in template
+}
+
 // WatchTopicValues values to populate the watch topic template
 type WatchTopicValues struct {
-	Name              string
-	Title             string
-	Scope             string
-	AgentResourceKind string
+	Name        string
+	Title       string
+	Description string
+	Kinds       []kindValues
 }
 
 // NewDiscoveryWatchTopic creates a WatchTopic template string.
 // Using a template instead of unmarshalling into a struct to avoid sending a request with empty fields
-func NewDiscoveryWatchTopic() string {
-	return `
-{
-	"group": "management",
-	"apiVersion": "v1alpha1",
-	"kind": "WatchTopic",
-	"name": "{{.Name}}",
-	"title": "{{.Title}}",
-	"spec": {
-		"filters": [
-			{
-				"group": "management",
-				"kind": "{{.AgentResourceKind}}",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["updated"]
-			},
-			{
-				"group": "management",
-				"kind": "AccessRequest",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["created", "updated"]
-			},
-			{
-				"group": "management",
-				"kind": "APIService",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["created", "updated", "deleted"]
-			},
-			{
-				"group": "management",
-				"kind": "APIServiceInstance",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["created", "updated", "deleted"]
-			},
-			{
-				"group": "catalog",
-				"kind": "Category",
-				"name": "*",
-				"type": ["created", "updated", "deleted"]
-			},
-			{
-				"group": "management",
-				"kind": "Credential",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["created", "updated"]
-			},
-			{
-				"group": "management",
-				"kind": "ManagedApplication",
-				"name": "*",
-				"scope": {
-					"kind": "Environment",
-					"name": "{{.Scope}}"
-				},
-				"type": ["created", "updated"]
-			}
-		],
-		"description": "Watch Topic used by a discovery agent for resources in the {{.Scope}} environment."
+func NewDiscoveryWatchTopic(name, scope, agentResourceKind string) WatchTopicValues {
+	return WatchTopicValues{
+		Name:        name,
+		Title:       name,
+		Description: fmt.Sprintf(desc, "discovery", scope),
+		Kinds: []kindValues{
+			{KindName: agentResourceKind, ScopeName: scope, EventTypes: updated},
+			{KindName: "Category", Group: "category", EventTypes: all},
```

Comment:
Can you please use the actual kind name, like `KindName: v1.AccessRequestGVK().Kind`